src/lib/SCEditor.js

```diff
@@ -1414,6 +1430,200 @@ define(function (require) {
 			});
 		};
 
+
+		/**
+		 * Adds a handle used to remap keys to make them more useful for code
+		 * input
+		 *
+		 * @private
+		 */
+		var addBodyCodeInputHandle = function (searchSelectors) {
+			$wysiwygBody.on('keydown', null,
+			{
+				'searchSelector': searchSelectors.join(',')
+			}, codeInputHandle);
+
+		};
+
+		/**
+		 * Adds a handle used to remap keys to make them more useful for
+		 * code input
+		 *
+		 * @private
+		 */
+		var codeInputHandle = function (e) {
+
+			if (e.ctrlKey || e.altKey || e.metaKey) {
+				return;
+			}
+
+			// For now, only Enter and Tab have defined executions
+			if (e.which !== 13 &&	// Enter
+				e.which !== 9		// tab
+			) {
+				return;
+			}
+
+			if (e.shiftKey && e.which !== 9) {
+				return;
+			}
+
+			rangeHelper.mergeTextNodesAtCaret();
```

Comment:
I've just tested it again and it is being called outside of code blocks. It runs until it reaches the `matches()` test so maybe if that was moved above `mergeTextNodesAtCaret()` it might fix it.